pystac_client/item_search.py

```diff
@@ -767,20 +772,16 @@ def item_collection_as_dict(self) -> Dict[str, Any]:
             Dict : A GeoJSON FeatureCollection
         """
         features = []
-        for page in self._stac_io.get_pages(
-            self.url, self.method, self.get_parameters()
-        ):
+        for page in self.pages_as_dicts():
             for feature in page["features"]:
                 features.append(feature)
                 if self._max_items and len(features) >= self._max_items:
                     feature_collection = {
                         "type": "FeatureCollection",
                         "features": features,
                     }
-                    call_modifier(self.modifier, feature_collection)
                     return feature_collection
```

Comment:
Do we need this? `pages_as_dicts` is limiting the number of items/features already.

Reply:
Good point. I mis-reasoned that we _did_ need this, but don't think we do. I'll remove.